### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by Ariges770**
Finished task 3

### CODE REVIEW

**Comment by Ariges770** [comment]
This is my final task 3 work

File: src/city.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def distance(self, other_city: City) -> int:
         :param other_city: a city to measure the distance to
         :return: the rounded-up distance in kilometers
         """
-        return round(distance.geodesic(self.coordinates, other_city.coordinates).km)
+        return math.ceil(distance.great_circle(self.coordinates, other_city.coordinates).km)
```
Comment (Ariges770) [inline src/city.py:65]: This was a bad mistake with my original task 1

Comment (Ariges770) [inline src/city.py:65]: This was a mistake with my original task 1 and I discovered the issue after properly reading the task requirements

File: src/itinerary.py [hunk 2]
```diff
@@ -20,32 +20,53 @@ def __init__(self, cities: list[City]) -> None:
...
 
     def min_distance_insert_city(self, city: City) -> None:
         """
         Inserts a city in the itinerary so that the resulting
         total distance of the itinerary is minimised.
         :param city: the city to insert
         :return: None.
-        """
-        #TODO
+        """
```
Comment (Ariges770) [inline src/itinerary.py:52]: Is this readable?

File: src/itinerary.py [hunk 3]
```diff
@@ -54,7 +75,16 @@ def __str__(self) -> str:
...
+            return "(0 km)"
+        for itinerary_index, city in enumerate(self.cities, start=1):
+            if itinerary_index == itinerary_len:
+                trip_str += f"{city.name} ({self.total_distance()} km)"
+                return trip_str
+            else:
+                trip_str += f"{city.name} -> "
 
 
 if __name__ == "__main__":
```
Comment (Ariges770) [inline src/itinerary.py:90]: The changes here aren't so important as they are just changes to the original Ed test cases

File: tests/conftest.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from city import City
```
Comment (Ariges770) [inline tests/conftest.py:1]: This rest is just unit testing, if you want I can teach you how it works to use at another time

File: src/itinerary.py [hunk 5]
```diff
@@ -20,32 +20,53 @@ def __init__(self, cities: list[City]) -> None:
...
         Returns the total distance (in km) of the itinerary, which is
         the sum of the distances between successive cities.
         :return: the total distance.
         """
-        #TODO
+        distance = 0
+        # Sum the distance from the next city ([1:]) to the preceding one, this avoids IndexError
+        for index, city in enumerate(self.cities[1:]):
+            distance += city.distance(self.cities[index])
+        return distance
```
Comment (mohanned-png) [inline src/itinerary.py:36]: perhaps change city to arrival_city and comment that self.cities[index] is departure city.

File: src/itinerary.py (hunk 2 above)
Comment (mohanned-png) [reply src/itinerary.py:52]: Nope. Add more comments to the try and except part of the for loop

File: src/itinerary.py [hunk 6]
```diff
@@ -54,7 +75,16 @@ def __str__(self) -> str:
 
         :return: a string representing the itinerary.
         """
-        #TODO
+        trip_str = ""
```
Comment (mohanned-png) [inline src/itinerary.py:78]: Comments here swell

File: src/itinerary.py (hunk 2 above)
Comment (mohanned-png) [reply src/itinerary.py:52]: I don't get the itin_city variable name

File: tests/conftest.py (hunk 4 above)
Comment (mohanned-png) [reply tests/conftest.py:1]: that will be cool

**Review by mohanned-png — APPROVED** [review]
Probably just comments and the variable name change I suggested

**Pull request merged**